File: chopper/models/game/hero.ts
import Card from "../card/card";
import Deck from "../card/deck";

export class Hero {
    public name: string = "";
    public description: string = "";
    public imageUrl: string = "";
    public health: number = 0;
    public move: string = "1";
    public attack: string = "1";
    public defense: string = "1";
    public backImageUrl: string = "";
    public deck: Card[];

    public constructor(json: string) { 
        let hero = JSON.parse(json);
        this.name = hero.name;
        this.description = hero.description;
        this.imageUrl = hero.imageUrl;
        this.health = hero.health;
        this.move = hero.move;
        this.attack = hero.attack;
        this.defense = hero.defense;
        this.backImageUrl = hero.backImageUrl;
        this.deck = hero.deck;
        this.handleCopies();
    }

    public handleCopies(): void {
        this.deck.forEach(card => {
            if(card.copies > 1) {
                for(let i = 1; i < card.copies; i++) {
                    let copy = new Card(card.name, card.description, card.frontImageUrl, this.backImageUrl);
                    this.deck.push(copy);
                }
                card.copies = 1;
            }
        });
    }
}

File: ChopperConsole/ChopperConsole/Models/Hero.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json;

namespace ChopperConsole.Models
{
    public class Card
    {
        public string name { get; set; }
        public string description { get; set; }
        public string frontImageUrl { get; set; }
        public int copies { get; set; }

        public Card()
        {
        }

        public Card(string nameIn, string descriptionIn, string frontImageUrlIn)
        {
            name = nameIn;
            description = descriptionIn;
            frontImageUrl = frontImageUrlIn;
            copies = 1;
        }
        public string GetText()
        {
            return (name + "\n " + description);
        }
    }

    public class Hero
    {
        public string name { get; set; }
        public string description { get; set; }
        public string image { get; set; }
        public string health { get; set; }
        public string attack { get; set; }
        public string move { get; set; }
        public string defense { get; set; }
        public string backImageUrl { get; set; }
        public List<Card> deck { get; set; }
        
        public static Hero FromJson(string json) {
            var hero = JsonSerializer.Deserialize<Hero>(json);
            if (hero == null) return null;
            hero.HandleCopies();
            return hero;
        }

        public void HandleCopies()
        {
            var newCards = new List<Card>();
            foreach (var card in deck.Where(card => card.copies > 1))
            {
                for (var i = 1; i < card.copies; i++)
                {
                    var copy = new Card(card.name, card.description, card.frontImageUrl);
                    newCards.Add(copy);
                }

                card.copies = 1;
            }
            deck.AddRange(newCards);
        }

        public void Shuffle()
        {
            var rng = new Random();  
            var n = deck.Count;  
            while (n > 1) {  
                n--;  
                var k = rng.Next(n + 1);  
                (deck[k], deck[n]) = (deck[n], deck[k]);
            }  
        }

        public string DrawCard()
        {
            var card = deck.First();
            deck.RemoveAt(0);
            deck.Add(card);
            return card.GetText();
        }
    }
}